packages/product-catalogue/src/components/ListView/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+import { Row, PageHeader, Col, Button, Table } from 'antd';
+import { loadProductCatalogue } from '../../helpers/dataLoaders';
+import { OpenSRPService } from '../../helpers/dataLoaders';
+import {
+  fetchProducts,
+  getProductArray,
+  getProductById,
+  ProductCatalogue,
+} from '../../ducks/productCatalogue';
+import { connect } from 'react-redux';
+import { ColumnsType } from 'antd/lib/table/interface';
+import { CatalogueLoading, columns } from './utils';
+import { Link, RouteComponentProps } from 'react-router-dom';
+import { Store } from 'redux';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import {
+  reducerName as ProductCatalogueReducerName,
+  ProductCatalogueReducer,
+} from '../../ducks/productCatalogue';
+import { BrokenPage, useHandleBrokenPage } from '@opensrp/react-utils';
+import { Helmet } from 'react-helmet';
+import { CATALOGUE_CREATE_VIEW_URL, RouteParams, TableColumnsNamespace } from '../../constants';
+import { ViewDetails } from '../ViewDetails';
+
+/** make sure product catalogue reducer is registered */
+reducerRegistry.register(ProductCatalogueReducerName, ProductCatalogueReducer);
+
+/** props for the productCatalogueList view */
+interface Props<T = ProductCatalogue> {
+  data: T[];
+  productUnderView: T | null;
+  columns: ColumnsType<T>;
+  service: typeof OpenSRPService;
+  fetchProductsCreator: typeof fetchProducts;
+}
+
+const defaultProps = {
+  data: [],
+  productUnderView: null,
+  columns: columns,
+  fetchProductsCreator: fetchProducts,
+  service: OpenSRPService,
+};
+
+export type ProductCatalogueListTypes = Props<ProductCatalogue> & RouteComponentProps<RouteParams>;
+
+/** component that renders product catalogue */
+
+const ProductCatalogueList = (props: ProductCatalogueListTypes) => {
+  const { service, data, columns, productUnderView, fetchProductsCreator } = props;
+  const [loading, setLoading] = useState<boolean>(data.length === 0);
+  const { broken, errorMessage, handleBrokenPage } = useHandleBrokenPage();
+
+  // see if we have a view-details product
+  const productId = props.match.params.productId ?? '';
+
+  useEffect(() => {
+    loadProductCatalogue(service, fetchProductsCreator)
+      .catch((err: Error) => handleBrokenPage(err))
+      .finally(() => setLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (loading) {
+    return <CatalogueLoading />;
+  }
+
+  if (broken) {
+    return <BrokenPage errorMessage={errorMessage} />;
+  }
+
+  const pageTitle = `Product Catalogue (${data.length})`;
```

Comment:
How are we dealing with making strings like this translatable?  I think this is important for EUSM?